src/pretix/control/views/checkin.py

```diff
@@ -69,6 +70,34 @@ def get_context_data(self, **kwargs):
                            or "subevent" in self.request.GET)
         return ctx
 
+    def post(self, request, *args, **kwargs):
+        for c in request.POST.getlist('checkin'):
+            created = False
+            op = OrderPosition.objects.select_related('item', 'variation', 'order', 'addon_to').get(
+                order__event=self.request.event,
+                order__code=c
+            )
+            if op.order.status == Order.STATUS_PAID:
+                ci, created = Checkin.objects.get_or_create(position=op, defaults={
+                    'datetime': now(),
+                })
+            if created:
+                op.order.log_action('pretix.control.views.checkin', data={
+                    'position': op.id,
+                    'positionid': op.positionid,
+                    'first': True,
+                    'datetime': now()
+                })
+            else:
+                op.order.log_action('pretix.control.views.checkin', data={
+                    'position': op.id,
+                    'positionid': op.positionid,
+                    'first': False,
+                    'datetime': now()
+                })
+
+        return self.get(request, *args, **kwargs)
```

Comment:
The problem with calling ``self.get`` here is, that two checkins will be created if the user now refreshs the page (as the form will be sent again). We should instead store a message to the user session and redirect back:

```
messages.success(request, _('The selected tickets have been marked as checked in.'))
return redirect(reverse('control:event.orders.checkins') + '?' + request.GET.urlencode())
```